web/libs/datamanager/src/components/MainView/GridView/ImagePreview.tsx

```diff
@@ -96,62 +100,84 @@ const ImagePreview = observer(({ task, field }: ImagePreviewProps) => {
-    const cursorX = e.clientX - rect.left;
-    const cursorY = e.clientY - rect.top;
-
-    // Zoom calculation
-    const newScale =
-      e.deltaY < 0
-        ? Math.min(scale * (1 + ZOOM_FACTOR), MAX_ZOOM) // Max zoom
-        : Math.max(scale * (1 - ZOOM_FACTOR), 1); // Min zoom
-
-    // Calculate zoom translation
-    const scaleDelta = newScale / scale;
-    // cursor - offset = cursor position relative to image; and that's the value being scaled.
-    // cursor position on a screen should stay the same, so we need to calculate new offset
-    // by scaling the distance to image edges and subtracting it from cursor position
-    const newX = cursorX - (cursorX - offset.x) * scaleDelta;
-    const newY = cursorY - (cursorY - offset.y) * scaleDelta;
-
-    setScale(newScale);
-    setOffset(constrainOffset({ x: newX, y: newY }));
-  };
+  }, []);
+
+  const handleWheel = useCallback(
+    (e: React.WheelEvent) => {
+      if (!containerRef.current || !imageLoaded) return;
+
+      const container = containerRef.current;
+      const rect = container.getBoundingClientRect();
+      const img = imageRef.current;
+      if (!img) return;
+
+      // Calculate cursor position relative to center
+      const cursorX = e.clientX - rect.left;
+      const cursorY = e.clientY - rect.top;
+
+      // Zoom calculation
+      const newScale =
+        e.deltaY < 0
+          ? Math.min(scale * (1 + ZOOM_FACTOR), MAX_ZOOM) // Max zoom
+          : Math.max(scale * (1 - ZOOM_FACTOR), 1); // Min zoom
+
+      // Calculate zoom translation
+      const scaleDelta = newScale / scale;
+      // cursor - offset = cursor position relative to image; and that's the value being scaled.
+      // cursor position on a screen should stay the same, so we need to calculate new offset
+      // by scaling the distance to image edges and subtracting it from cursor position
+      const newX = cursorX - (cursorX - offset.x) * scaleDelta;
+      const newY = cursorY - (cursorY - offset.y) * scaleDelta;
+
+      setScale(newScale);
+      setOffset(constrainOffset({ x: newX, y: newY }));
+    },
+    [imageLoaded, offset, scale, constrainOffset],
+  );
 
-  const handleMouseDown = (e: React.MouseEvent) => {
-    if (!containerRef.current || scale <= 1) return;
+  const handleMouseMove = useCallback(
+    (e: MouseEvent) => {
+      if (!containerRef.current || !imageRef.current) return;
 
-    setIsDragging(true);
-    setDragAnchor({ x: e.clientX, y: e.clientY });
-    setStartOffset({ x: offset.x, y: offset.y });
-  };
+      const { x: oldX, y: oldY } = dragParams.current.dragAnchor;
+      const { x: offsetX, y: offsetY } = dragParams.current.startOffset;
+      const newX = e.clientX - oldX;
+      const newY = e.clientY - oldY;
 
-  const handleMouseMove = (e: React.MouseEvent) => {
-    if (!isDragging || !containerRef.current || !imageRef.current) return;
+      setOffset(constrainOffset({ x: offsetX + newX, y: offsetY + newY }));
```

Comment:
I see the logic is changing here - and it seems reasonable but feels like we should have a test added to make sure the behaviour is consistent with expectations

Reply:
oh i see - i think the change in order for the functions threw me off a little but it looks like a refactor - not a logic change - disregard by previous comment